WordPress/gutenberg
# Edit block style variations from global styles

## Why?


Closes #44417.

This is an initial approach to editing block style variations in global styles. It's still a work in progress and not the complete implementation for #44417.

## How?


Currently this PR only allows to edit existing variations, which works to a limited extent, given some of the classic core block variations style elements not yet available as block supports, or aren't added to every block. It also leverages the existing global styles interface for now, so the "variations" section appears as if it were another block support. 

#### Limitations

* Style variations can only make use of whatever the block supports, so their scope is much more limited than the classic CSS variations.

#### Todoes

* It should be possible to add new style variations to any block;
* "Variations" should be a section of its own, as shown in the designs in #44417.

## Testing Instructions





1. In Global Styles, navigate to "Blocks" and then into a block that has variations, e.g. Button, Image, or Site Logo.
2. Check that a "Variations" section is rendered inside the Block screen and that clicking on that section brings up the Variations screen;
3. Check that inside the Variations screen all core variations for the block are rendered;
4. Clicking into the variations, it should be possible to change their styles and those changes should only affect that specific variation.

### Testing Instructions for Keyboard


Global Styles can be accessed from the Site Editor header by clicking the "Styles" button. Unfortunately keyboard navigation to the sidebar is broken, so it is necessary to tab 5-6 times after clicking "Styles" to actually arrive in the sidebar. (The first button in the sidebar is called "Open Style Book"; from then on tabbing a few more times should take us to the "Blocks" section)

## Screenshots or screencast 

<img width="275" alt="Screenshot 2022-12-20 at 2 19 08 pm" src="https://user-images.githubusercontent.com/8096000/208573637-3359b07c-1bed-48f7-9cce-1b883ad35755.png">
<img width="281" alt="Screenshot 2022-12-20 at 2 19 37 pm" src="https://user-images.githubusercontent.com/8096000/208573646-cd199ab5-24c3-4413-a708-300035fc0b3f.png">
<img width="282" alt="Screenshot 2022-12-20 at 2 19 46 pm" src="https://user-images.githubusercontent.com/8096000/208573654-0db6744e-940b-4bc4-91ff-cefaed9b89d4.png">
